### ISSUE TITLE
Incorrect unquotation of type lambda trick

### ISSUE THREAD

**Issue by notxcain** [scalameta/scalameta#565]
This  
```scala
q"""
  def foo[F[_]]: Unit = ???
  foo[({ type T[A] = Either[Int, A] })#T]
"""
```
is being unquoted to this (notice missing parenthesis):  
```scala
def foo[F[_]]: Unit = ???
foo[{ type T[A] = Either[Int, A] }#T]
```
and does not compile.

Not sure if it is expected behavior, for me it wasn't though :shipit:

**Comment by xavier-fernandez** [comment]
Will work on this issue this weekend :)

**Comment by xavier-fernandez** [comment]
The commit #633 fixed this issue, this ticket can be closed :)

**Comment by notxcain** [comment]
Awesome, thanks!

**Comment by notxcain** [comment]
Hi! I'm on 1.6.0-635 and I'm still getting this
```scala
scala> q"def foo[A]: Foo[({ type T[F[_]] = Bar[A, F] })#T, Unit] = null".show[Syntax]
res0: String = def foo[A]: Foo[{ type T[F[_]] = Bar[A, F] }#T, Unit] = null
```

**Comment by xavier-fernandez** [comment]
Will check this issue tomorrow

**Comment by xavier-fernandez** [comment]
@notxcain I am not able to reproduce this bug in master, I suspect '1.6.0-635' do not haves the fix implemented yet :(

https://github.com/xavier-fernandez/scalameta/tree/recheck-%23565

**Comment by xeno-by** [comment]
@notxcain Have you tried 1.6.0-633?

**Comment by notxcain** [comment]
@xavier-fernandez @xeno-by my bad. I was confused with version naming. It is fixed on `1.6.0-633`. Thank you!

**Comment by xavier-fernandez** [comment]
@notxcain no problem :)
